cmd/release/cmd/stats.go

```diff
@@ -58,128 +48,52 @@ var statsCmd = &cobra.Command{
 			return errors.New("end date before start date")
 		}
 
-		githubToken := os.Getenv("GITHUB_TOKEN")
-
 		ctx := context.Background()
-		client := repository.NewGithub(ctx, githubToken)
+		client := repository.NewGithub(ctx, rootConfig.Auth.GithubToken)
 
 		s := spinner.New(spinner.CharSets[31], 100*time.Millisecond)
 		s.HideCursor = true
+		s.Writer = os.Stderr
 		s.Start()
 
-		var total int
-		data := make(map[int]relStats)
-		captains := make(map[string]int)
-
-		lo := github.ListOptions{
-			PerPage: 100,
+		sd, err := release.Stats(ctx, client, from, to, repoToOwner[*repo], *repo)
+		if err != nil {
+			return err
 		}
-		for {
-			releases, resp, err := client.Repositories.ListReleases(ctx, repoToOwner[*repo], *repo, &lo)
-			if err != nil {
-				return err
-			}
-
-			for _, release := range releases {
-				releaseDate := release.GetCreatedAt().Time
-				if releaseDate.After(from) && (releaseDate.Before(to) || releaseDate.Equal(to)) {
-					total++
-
-					if _, ok := data[int(release.CreatedAt.Year())]; !ok {
-						data[int(release.CreatedAt.Year())] = relStats{
-							count: 1,
-							monthly: map[time.Month]monthly{
-								release.CreatedAt.Month(): {
-									count: 1,
-									captains: []string{
-										*release.Author.Login,
-									},
-									tags: []string{
-										*release.Name,
-									},
-								},
-							},
-						}
-						continue
-					}
-
-					rs := data[int(release.CreatedAt.Year())]
-					rs.count++
-
-					mon := rs.monthly[release.CreatedAt.Month()]
-					mon.count++
-					mon.captains = append(mon.captains, *release.Author.Login)
-					mon.tags = append(mon.tags, *release.Name)
-
-					rs.monthly[release.CreatedAt.Month()] = mon
-
-					data[int(release.CreatedAt.Year())] = rs
-
-					if release.Author.Login != nil {
-						if _, ok := captains[*release.Author.Login]; !ok {
-							captains[*release.Author.Login]++
-							continue
-						}
-						captains[*release.Author.Login]++
-					}
-				}
-			}
-
-			if resp.NextPage == 0 {
-				break
-			}
-			lo.Page = resp.NextPage
+
+		if format == nil || *format == "" {
+			*format = "json"
 		}
 
-		s.Stop()
+		var b []byte
 
-		for year := range data {
-			fmt.Printf("\n%d:\n", year)
-
-			months := make([]int, 0, len(data[year].monthly))
-			for k := range data[year].monthly {
-				months = append(months, int(k))
-			}
-			sort.Ints(months)
-
-			for _, m := range months {
-				mon := time.Month(m)
-				tmp := data[year].monthly[mon]
-				tmp.captains = dedup(tmp.captains)
-				data[year].monthly[mon] = tmp
-				captains := strings.Join(data[year].monthly[mon].captains, ", ")
-				tags := strings.Join(data[year].monthly[mon].tags, ", ")
-				fmt.Printf("  %-9s\n    Count: %3d\n    Captains: %s\n    Tags: %s\n",
-					mon, data[year].monthly[mon].count, captains, tags)
-			}
+		switch *format {
+		case "json":
+			b, err = json.Marshal(sd)
+		case "yaml":
+			b, err = yaml.Marshal(sd)
+		default:
+			return errors.New("unrecognized format")
```

Comment:
Just a small suggestion.

```suggestion
			return fmt.Errorf("unrecognized format '%s'", *format)
```

Reply:
Given the cli ux the user specified format is directly above the error message.